sympy/matrices/repmatrix.py

```diff
@@ -332,6 +344,17 @@ def _eval_row_join(self, other):
     def _eval_extract(self, rowsList, colsList):
         return self._fromrep(self._rep.extract(rowsList, colsList))
 
+    @overload
+    def __getitem__(self, key: tuple[int, int]) -> Expr: ...
+    @overload
+    def __getitem__(self, key: int) -> Expr: ...
+    @overload
+    def __getitem__(self, key: tuple[int, slice]) -> Self: ...
+    @overload
+    def __getitem__(self, key: tuple[slice, int]) -> Self: ...
+    @overload
+    def __getitem__(self, key: tuple[slice, slice]) -> Self: ...
+
     def __getitem__(self, key):
         return _getitem_RepMatrix(self, key)
```

Comment:
The important thing in this PR is this. Typecheckers get very confused about what `M[0,0]` does for a matrix.